compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -298,6 +302,19 @@ impl Context {
         block_id
     }
 
+    /// Get the next BlockId for internal memory
+    /// used during ACIR generation.
+    /// This is useful for referencing information that can
+    /// only be computed dynamically, such as the type structure
+    /// of non-homogenous arrays.
+    fn internal_block_id(&mut self) -> BlockId {
+        let block_id = BlockId(self.max_block_id);
+        self.max_block_id += 1;
```

Comment:
Going to add a separate map here so we are not repeatedly initializing these type element size arrays

Reply:
I have added an `internal_memory_blocks` map. This reduced the amount of block ids in the example test under `nested_array_dynamic` from about 90 to about 36. However, the final optimized ACIR only has one less memory init.